### PR TITLE
A sample to enable OSPatching v2.0

### PR DESCRIPTION
**Pull request by bingosummer**
Update parameters and change to 201 template

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@bingosummer__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: 201-ospatching-extension-on-ubuntu/azuredeploy.json [hunk 1]
```diff
@@ -2,17 +2,12 @@
...
-      "metadata": {
-        "description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
-      }
-    },
     "location": {
       "type": "string",
       "metadata": {
         "description": "The location where the Virtual Machine will be deployed."
       },
+      "defaultValue": "West US",
```
Comment (singhkay) [inline 201-ospatching-extension-on-ubuntu/azuredeploy.json]: Please remove the default value and let the user choose it

File: 201-ospatching-extension-on-ubuntu/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Simple deployment of an Ubuntu VM with OS Patching extension
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-ospatching-extension-on-ubuntu%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+
+Built by: [Thomas Shao](https://github.com/thomas1206)
```
Comment (singhkay) [inline 201-ospatching-extension-on-ubuntu/README.md]: Built by and the table of parameters below should be removed. We will soon render readme's on Azure.com and these will end up being duplicated.

File: 201-ospatching-extension-on-ubuntu/azuredeploy.json [hunk 3]
```diff
@@ -98,17 +106,41 @@
...
     }
   },
   "variables": {
+    "location": "[parameters('location')]",
+    "vmSize": "Standard_D1",
     "imagePublisher": "Canonical",
     "imageOffer": "UbuntuServer",
+    "ubuntuOSVersion": "14.04.2-LTS",
     "OSDiskName": "osdiskforospatching",
     "nicName": "myVMNic",
```
Comment (singhkay) [inline 201-ospatching-extension-on-ubuntu/azuredeploy.json]: Please use a name that follows this scheme for these variables - templateScenarioResourceName

File: 201-ospatching-extension-on-ubuntu/azuredeploy.json [hunk 4]
```diff
@@ -119,7 +151,7 @@
     "publicIPAddressName": "myPublicIP",
```
Comment (singhkay) [inline 201-ospatching-extension-on-ubuntu/azuredeploy.json]: Please use a name that follows this scheme for these variables - templateScenarioResourceName

File: 201-ospatching-extension-on-ubuntu/azuredeploy.json [hunk 5]
```diff
@@ -119,7 +151,7 @@
...
     "publicIPAddressType": "Dynamic",
     "vmStorageAccountContainerName": "vhds",
-    "vmName": "MyOSPatchingVM",
+    "vmName": "MyUbuntuVM",
```
Comment (singhkay) [inline 201-ospatching-extension-on-ubuntu/azuredeploy.json]: Please use a name that follows this scheme for these variables - templateScenarioResourceName

File: 201-ospatching-extension-on-ubuntu/azuredeploy.json [hunk 6]
```diff
@@ -119,7 +151,7 @@
...
     "virtualNetworkName": "MyVNET",
```
Comment (singhkay) [inline 201-ospatching-extension-on-ubuntu/azuredeploy.json]: Please use a name that follows this scheme for these variables - templateScenarioResourceName

File: 201-ospatching-extension-on-ubuntu/azuredeploy.json [hunk 7]
```diff
@@ -98,17 +106,41 @@
...
+    "storageAccountKey": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "The access key of storage account."
       }
     }
   },
   "variables": {
+    "location": "[parameters('location')]",
```
Comment (singhkay) [inline 201-ospatching-extension-on-ubuntu/azuredeploy.json]: What purpose does this variable serve compared to the parameter?

File: 201-ospatching-extension-on-ubuntu/azuredeploy.json [hunk 8]
```diff
@@ -129,7 +161,7 @@
       "type": "Microsoft.Storage/storageAccounts",
       "name": "[parameters('newStorageAccountName')]",
       "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline 201-ospatching-extension-on-ubuntu/azuredeploy.json]: Can you make the api version a variable so it's easier top move to newer version headers?

**Comment by singhkay** [comment]
@bingosummer Looks good, thanks for fixing!

**Pull request merged**